ui/app/AppLayouts/Profile/views/NotificationsView.qml

```diff
@@ -137,29 +122,15 @@ SettingsContentBase {
 
                 components: [
                     StatusIcon {
-                        visible: model.customized
-                        icon: "next"
-                        color: Theme.palette.baseColor1
+                        icon: model.customized ? "next" : "add"
+                        color: model.customized ? Theme.palette.baseColor1 : Theme.palette.primaryColor1
                         StatusMouseArea {
                             anchors.fill: parent
                             cursorShape: Qt.PointingHandCursor
-                            onClicked: {
-                                exemptionNotificationsModal.open(model)
-                            }
+                            onClicked: exemptionNotificationsModal.createObject(root, {item: model}).open()
```

Comment:
There is bad practice used there: `model` object which is managed by given delegate's instance is passed outside to component not managed by a delegate. It may lead to unexpected behavior in some corner cases. E.g. in full window mode with listview occupying full height, the modal is invoked for first or last visible delegate. Then, with the modal opened, the window is resized to much smaller size. It may cause delegate to be destroyed, among with the `model` object passed to modal.

The simplest option would be probably to change `ExemptionNotificationsModal` to take all necessary values separately instead of one `model` object assigned to `item` var property.

Reply:
Hmm, is it likely to happen since the modal basically makes a copy of the `item`? :)

```qml
        property bool muteAllMessages: root.item.muteAllMessages
        property string personalMentions: root.item.personalMentions
        property string globalMentions: root.item.globalMentions
        property string otherMessages: root.item.otherMessages
```

Reply:
But I agree, in general this is not a nice approach

Reply:
Fixed, pls have a second look @micieslak